validphys2/src/validphys/n3fit_data_utils.py

```diff
@@ -1,168 +1,107 @@
-
-    #Returns:
-        - `[dataset_dict]`: a (len 1 list of) dictionary with:
-            - dataset: full dataset object from which all data has been extracted
-            - hadronic: is hadronic?
-            - operation: operation to perform on the fktables below
-            - fktables : a list of dictionaries with the following items
-                - ndata: number of data points
-                - nonzero/nbasis: number of PDF entries which are non zero
-                - basis: array containing the information of the non-zero PDF entries
-                - nx: number of points in the xgrid (1-D), i.e., for hadronic the total number is nx * nx
-                - xgrid: grid of x points
-                - fktable: 3/4-D array of shape (ndata, nonzero, nx, (nx))
-
-    instead of the dictionary object that model_gen needs
+    Python version of the libNNPDF dataset
+    to be merged with the product of the new CommonData dataset
+
+    Parameters
+    ----------
+        name: str
+            name of the dataset
+        fktables_data: list(:py:class:`validphys.coredata.FKTableData`)
+            list of coredata fktable objects
+
+        operation: str
+            operation to be applied to the fktables in the dataset, default "NULL"
+        frac: float
+            fraction of the data to enter the training set
+        training_mask: bool
+            training mask to apply to the fktable
     """
-    how_many = dataset_c.GetNSigma()
-    dict_fktables = []
-    for i in range(how_many):
-        fktable = dataset_c.GetFK(i)
-        dict_fktables.append(fk_parser(fktable, dataset_c.IsHadronic()))
-
-    dataset_dict = {
-        "fktables": dict_fktables,
-        "hadronic": dataset_c.IsHadronic(),
-        "operation": dataset_spec.op,
-        "name": dataset_c.GetSetName(),
-        "frac": dataset_spec.frac,
-        "ndata": dataset_c.GetNData(),
-    }
-
-    return [dataset_dict]
-
-
-def common_data_reader_experiment(experiment_c, experiment_spec):
-    """
-    Wrapper around the experiments. Loop over all datasets in an experiment,
-    calls common_data_reader on them and return a list with the content.
-
-    # Arguments:
-        - `experiment_c`: c representation of the experiment object
-        - `experiment_spec`: python representation of the experiment object
-
-    # Returns:
-        - `[parsed_datasets]`: a list of dictionaries output from `common_data_reader_dataset`
-    """
-    parsed_datasets = []
-    for dataset_c, dataset_spec in zip(experiment_c.DataSets(), experiment_spec.datasets):
-        parsed_datasets += common_data_reader_dataset(dataset_c, dataset_spec)
-    return parsed_datasets
-
 
-def positivity_reader(pos_spec):
+    # NOTE:
+    # This class tries to be compatible the libNNPDF dataset class
+    # after the class is moved to python, this can inherit from dataset
+    # and the dataset should know how to generate its "fittable" version
```

Comment:
```suggestion
    # after the class is moved to python, this can inherit from nnpdf.DataSet
    # and the dataset should know how to generate its "fittable" version
```

Not really sure what you mean here

Reply:
The class `FittableDataset` tries to be equal to `libNNPDF` since we still have `libNNPDF` stuff (due to commondata). Once `libNNPDF` is no longer used, this can just inherit from the vp dataset.